src/io_callbacks.zig

```diff
@@ -87,10 +116,12 @@ pub const IO = struct {
             }
             if (completed < cqes.len) break;
         }
+        return timeout_complete;
     }
 
-    fn flush_submissions(self: *IO, wait: bool) !void {
+    fn flush_submissions(self: *IO, wait: bool) !bool {
         var wait_nr: u32 = if (wait) 1 else 0;
+        var timeout_complete = false;
         while (true) {
             wait_nr = std.math.min(self.queued + self.submitted, wait_nr);
             _ = self.ring.submit_and_wait(wait_nr) catch |err| switch (err) {
```

Comment:
Ignoring this value seems wrong, and when I assert that the returned value is equal to `self.queued` the assertion fails. I think this could actually have been the root cause of the original issues we were seeing, but I'm not entirely sure what exactly causes this to happen or what to do about it.

Reply:
It seems that we don't actually have a need to check `pending` anywhere so I've just removed the counters entirely.

Reply:
That's also 100%. I had thought we could use it to assert that the event loop drives IO without sitting idle.